feat: add `destructuredArrayIgnorePattern` option in `no-unused-vars` (#15649)
eslint/eslint: docs/rules/no-unused-vars.md

foo();
```

### destructuredArrayIgnorePattern

> The `destructuredArrayIgnorePattern` option specifies exceptions not to check for usage: destructuring assignments from an array whose names match a regexp pattern. For example, variables whose names begin with an underscore.

Reviewer: ```suggestion
The `destructuredArrayIgnorePattern` option specifies exceptions not to check for usage: elements of array destructuring patterns whose names match a regexp pattern. For example, variables whose names begin with an underscore.
```

foo();
```

### destructuredArrayIgnorePattern

The `destructuredArrayIgnorePattern` option specifies exceptions not to check for usage: elements of array destructuring patterns whose names match a regexp pattern. For example, variables whose names begin with an underscore.

Examples of **correct** code for the `{ "destructuredArrayIgnorePattern": "^_" }` option:

```js
/*eslint no-unused-vars: ["error", { "destructuredArrayIgnorePattern": "^_" }]*/

const [a, _b, c] = ["a", "b", "c"];
console.log(a+c);

const { x: [_a, foo] } = bar;
console.log(foo);

function baz([_c, x]) {
    x;
}
baz();

> 

Reviewer 2: i'm confused. in an array pattern, you never need an unused variable, you just omit the binding - ie, `[, x]` etc. Why would you ever want an exception here?

Reviewer: The rationale for this option is in Additional Comments of the original post in https://github.com/eslint/eslint/issues/15611.

This is optional behavior, unused variables in array patterns are not ignored by default. Users who prefer elisions can simply not enable this option.

Reviewer 2: So, this is a style option, added after the prohibition on such things?

foo();
```

### destructuredArrayIgnorePattern

The `destructuredArrayIgnorePattern` option specifies exceptions not to check for usage: elements of array destructuring patterns whose names match a regexp pattern. For example, variables whose names begin with an underscore.

Examples of **correct** code for the `{ "destructuredArrayIgnorePattern": "^_" }` option:

```js
/*eslint no-unused-vars: ["error", { "destructuredArrayIgnorePattern": "^_" }]*/

const [a, _b, c] = ["a", "b", "c"];
console.log(a+c);

const { x: [_a, foo] } = bar;
console.log(foo);

function baz([_c, x]) {
    x;
}
baz();

function test({p: [_q, r]}) {
    r;
}
test();

> let _m, n;
> 
> foo.forEach(item => {
>     [_m, n] = item;
>     console.log(n);
> });

Reviewer: ```suggestion
let _m, n;
foo.forEach(item => {
    [_m, n] = item;
    console.log(n);
});
```

I think it's better to remove this blank line since all these lines are 1 example. That would be consistent with other examples in this code block (blank line starts a new example).
